### PR TITLE
amazon-import post-processor: support license_type

### PR DESCRIPTION
**Pull request by dayglojesus**
Implements a new config option for the amazon-import post-processor
allowing users to specify a LicenseType during the import process.

Closes: #4631

### CODE REVIEW

**Comment by mwhooker** [comment]
That's awesome, thank you! We'll get this in the next release.

Would you mind also adding docs to https://github.com/mitchellh/packer/blob/master/website/source/docs/post-processors/amazon-import.html.md ?

**Comment by dayglojesus** [comment]
Done. Thanks @mwhooker!
https://github.com/mitchellh/packer/pull/4636

**Comment by dayglojesus** [comment]
@mwhooker yeah, sorry about that, I was going to push to the same branch but the PR was already merged, so I got confused.

**Comment by mwhooker** [comment]
hmm, not sure what happened there then. I haven't merged it yet

**Comment by dayglojesus** [comment]
I was prob looking at the wrong PR. Sorry.

**Comment by kwilczynski** [comment]
@dayglojesus thank you for this! I ended up in a situation recently were I needed this to be handled, and having it in Packer is a huge time saver!

**Comment by mwhooker** [comment]
I think this needs a quick rebase

**Comment by dayglojesus** [comment]
@mwhooker Sorry about the wait, but I think that should do it.

**Comment by dayglojesus** [comment]
@mwhooker guess this didn't make the `1.0.0` release :(

**Comment by mwhooker** [comment]
@dayglojesus 

> @mwhooker mwhooker added this to the v1.1 milestone on Mar 7

**Comment by dayglojesus** [comment]
thanks @mwhooker I was not aware that the `1.0.0` release was imminent

**Comment by spion06** [comment]
+1 to wanting this. I had a PR that I was JUST about to hit submit on until I saw this one. I can provide my manual testing if that'll help get it merged quicker :)

File: post-processor/amazon-import/post-processor.go [hunk 1]
```diff
@@ -63,6 +64,10 @@ func (p *PostProcessor) Configure(raws ...interface{}) error {
 		p.config.S3Key = "packer-import-{{timestamp}}.ova"
 	}
 
+	if p.config.LicenseType == "" {
+		p.config.LicenseType = "BYOL"
```
Comment (mwhooker) [inline post-processor/amazon-import/post-processor.go]: is byol the default license type?

that is, if no license type is configured, are images automatically imported as BYOL? I'm trying to understand if this changes the behavior at all.

Comment (dayglojesus) [inline post-processor/amazon-import/post-processor.go]: The default is actually platform dependent: Linux is BYOL, Windows is AWS.

Comment (mwhooker) [inline post-processor/amazon-import/post-processor.go]: thanks! Can we change this then so we don't set a license type unless it's explicitly given?

Comment (kwilczynski) [inline post-processor/amazon-import/post-processor.go]: So, either detect the platform or document it very, very well :)

Comment (dayglojesus) [inline post-processor/amazon-import/post-processor.go]: Of course, if you light up a Windows box and you use the default AWS licensing, I hope you remembered to pack your wallet. 😁

Comment (spion06) [inline post-processor/amazon-import/post-processor.go]: Funny enough. When I came back to my local build after it got farther in the import it received and error saying AWS licensing isn't valid for the 'Linux' platform.

```
1495058104,,ui,say,==> file: Running post-processor: manifest
1495058104,,ui,error,Build 'file' errored: 1 error(s) occurred:\n\n* Post-processor failed: Import task import-ami-fg3rhnba failed: ClientError: The chosen license type (AWS) is incompatible with the chosen platform type (Linux).
``` 

That's what I get for not testing with the CLI first.

**Comment by dayglojesus** [comment]
@mwhooker Sorry for the delay. I think this should do the trick.

Side Note: I am not adept with Golang, but a colleague and I bashed our heads together on using the [http://docs.aws.amazon.com/sdk-for-go/api/service/ec2/#ImportImageInput](ImportImageInput) method. We soon realized that this method was not actually in the SDK that we fetched when we did a `go get`. Instead, we simply opted to add the key to the struct.

**Comment by mwhooker** [comment]
Sorry, this still sets the license type if it's empty. We should only change the license type if it's requested by the user

**Comment by dayglojesus** [comment]
@mwhooker once again, fixed. sorry.

**Comment by mwhooker** [comment]
thanks!

**Pull request merged**